sct-commons/src/test/java/org/lfenergy/compas/sct/commons/scl/ied/LN0AdapterTest.java

```diff
@@ -858,4 +858,29 @@ void getFCDAs_should_throw_Exception_when_DataSet_not_present() {
                 .hasMessage("Control Block smv1 not found in /LN0");
     }
 
+    @Test
+    void streamControlBlocks_should_return_all_GSEControlBlocks(){
+        // Given
+        IEDAdapter iedAdapter = mock(IEDAdapter.class);
+        TLDevice tlDevice = new TLDevice();
+        LDeviceAdapter lDeviceAdapter = mock(LDeviceAdapter.class);
+        when(lDeviceAdapter.hasDataSetCreationCapability(any())).thenReturn(true);
+        when(lDeviceAdapter.hasControlBlockCreationCapability(any())).thenReturn(true);
+        when(lDeviceAdapter.getParentAdapter()).thenReturn(iedAdapter);
+        LN0 ln0 = new LN0();
+        tlDevice.setLN0(ln0);
+        when(lDeviceAdapter.getCurrentElem()).thenReturn(tlDevice);
+        LN0Adapter ln0Adapter = new LN0Adapter(lDeviceAdapter, ln0);
+        ln0Adapter.createDataSetIfNotExists("datSet1", ControlBlockEnum.GSE);
+        ln0Adapter.createDataSetIfNotExists("datSet2", ControlBlockEnum.SAMPLED_VALUE);
+        ln0Adapter.createControlBlockIfNotExists("cbNameGSE", "cbId1", "datSet1", ControlBlockEnum.GSE);
+        ln0Adapter.createControlBlockIfNotExists("cbNameSMV", "cbId2", "datSet2", ControlBlockEnum.SAMPLED_VALUE);
+        // When
+        Stream<ControlBlockAdapter> result = ln0Adapter.streamControlBlocks(ControlBlockEnum.GSE);
+        // Then
+        assertThat(result)
```

Comment:
Could add hasSize(1)

Reply:
OK.
FYI it is tested implicitly with `.containsExactly("cbNameGSE")`.